cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java

```diff
@@ -168,6 +168,56 @@ public void configure(Map<String, ?> config) {
     ensureTopicsCreated(config);
   }
 
+  /**
+   * Retrieve the desired replication factor of sample store topics.
+   *
+   * @param config The configurations for Cruise Control.
+   * @param adminClient The adminClient to send describeCluster request.
+   * @return Desired replication factor of sample store topics, or {@link null} if failed to resolve replication factor.
+   */
+  protected Short sampleStoreTopicReplicationFactor(Map<String, ?> config, AdminClient adminClient) {
+    if (_sampleStoreTopicReplicationFactor == null) {
+      short numberOfBrokersInCluster;
+      try {
+        numberOfBrokersInCluster = (short) adminClient.describeCluster().nodes().get(CLIENT_REQUEST_TIMEOUT_MS,
+                                                                                     TimeUnit.MILLISECONDS).size();
+      } catch (InterruptedException | ExecutionException | TimeoutException e) {
+        LOG.warn("Auto creation of sample store topics failed due to failure to describe cluster.", e);
+        return null;
+      }
+      if (numberOfBrokersInCluster <= 1) {
+        throw new IllegalStateException(String.format("Kafka cluster has less than 2 brokers (brokers in cluster=%d, zookeeper.connect=%s)",
+                                                      numberOfBrokersInCluster, config.get(ExecutorConfig.ZOOKEEPER_CONNECT_CONFIG)));
+      }
+
+      return (short) Math.min(DEFAULT_SAMPLE_STORE_TOPIC_REPLICATION_FACTOR, numberOfBrokersInCluster);
+    }
+
+    return _sampleStoreTopicReplicationFactor;
+  }
+
+  /**
+   * Creates the given topic if it does not exist.
+   *
+   * @param adminClient The adminClient to send createTopics request.
+   * @param topicToBeCreated A wrapper around the topic to be created.
+   * @return {@code false} if the topic to be created already exists, {@code true} otherwise.
+   */
+  protected static boolean createTopic(AdminClient adminClient, NewTopic topicToBeCreated) {
+    try {
+      CreateTopicsResult createTopicsResult = adminClient.createTopics(Collections.singletonList(topicToBeCreated));
+      createTopicsResult.values().get(topicToBeCreated.name()).get(CruiseControlMetricsUtils.CLIENT_REQUEST_TIMEOUT_MS,
+                                                                   TimeUnit.MILLISECONDS);
+      LOG.info("Topic {} has been created.", topicToBeCreated.name());
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      if (e.getCause() instanceof TopicExistsException) {
+        return false;
+      }
+      LOG.warn("Unable to create topic {}.", topicToBeCreated.name(), e);
```

Comment:
If we are unable to create topic here, shouldn't we throw out exception? I think if this topic does not exist and is fail to be created,  `KafkaSampleStore` is unable to work.